spec/MyArraySpec.js

```diff
@@ -360,6 +360,17 @@ describe('MyArray', function () {
             var expected = MyArray.of('1,0', '2,1', '10,2', '3,3', 'abc,4', '4,5');
             expect(receivedElements.equals(expected)).toEqual(true);
         });
+
+        it('preserves original array', function() {
+            var array = MyArray.of(1, '2', 10, 3, 'abc', 4);
+            var receivedElements = new MyArray();
+
+            array.forEach(function (element, index) {
```

Comment:
I think to break the code in #3 you would need to `return 'something';` here :)

Reply:
`function() {}` should return undefined, so that each element ends up undefined in the concrete case. I'll test that out and get back to you :)

Reply:
`forEach` there has this:

```javascript
var r = fn(this.get(i), i);
if(r != undefined) {
    this.set(i, r);
}
```

So this will not modify anything if the function returns `undefined` (or something that coerces down to it). Including function without return.

Reply:
Good catch!

Reply:
Thanks for pointing this out! I didn't consider the modifications the array would incorrectly have.